### ISSUE TITLE
Uninformative error message when filtering empty tasks

### ISSUE THREAD

**Issue by mb706** [mlr-org/mlr3filters#39]
Filtering a 0-row task
```r
> tsk = mlr_tasks$get("iris")$filter(integer(0))
> mlr3featsel::FilterVariance$new()$calculate(tsk)
Error in mlr3featsel::FilterVariance$new()$calculate(tsk) : 
  Assertion on 'fv' failed: Contains missing values (element 1).
```
Filtering a 0-column task
```r
> tsk = mlr_tasks$get("iris")$select(integer(0))
> mlr3featsel::FilterVariance$new()$calculate(tsk)
Error in bmerge(i, x, leftcols, rightcols, xo, roll, rollends, nomatch,  : 
  x.'id' is a character column being joined to i.'V1' which is type 'integer'. Character 
```
The first one should probably give a more informative error message (the filtering may happen as part of a longer pipeline where the user would not immediately know that an empty task was given). The second one should imho not even throw an error.

**Comment by mb706** [comment]
At least some of the filters also seem to crash when there is only one row of data.

**Comment by mllg** [comment]
Fixed in 498d185.
